src/main/com/algos/src/datastructures/caches/LRUCache.java

```diff
@@ -89,18 +79,15 @@ private void moveNodeToLast(Entry<K, V> entry) {
     public void put(K key, V value) {
         if (data.containsKey(key)) {
             final Entry<K, V> existingEntry = data.get(key);
-            existingEntry.setValue(value);
             moveNodeToLast(existingEntry);
+            existingEntry.setValue(value); 
             return;
         }
-        Entry<K, V> newEntry;
-        if (data.size() == cap) {
+        Entry<K, V> newEntry = new Entry<>(); 
+        if (data.size() >= cap) {
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/rishabhbohra3/Algo-tutorial-test/pull/6#issuecomment-2196728573>#cbf3aa</a><span> - 06/28/2024, 11:53 am</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The condition 'if (data.size() >= cap)' is correct for checking if the cache is full, but the subsequent logic for eviction and entry creation can lead to potential issues. Specifically, the line 'newEntry = evict();' reassigns 'newEntry' to the evicted entry, which can cause confusion and potential bugs if not handled carefully.
</div>
<div id="fix">
<b>Fix:</b> Separate the eviction logic from the new entry creation to avoid reassigning 'newEntry'. This will make the code clearer and reduce the risk of bugs.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -86,10 +86,12 @@
      Entry<K, V> newEntry = new Entry<>();
      if (data.size() >= cap) {
          Entry<K, V> evictedEntry = evict();
          data.remove(evictedEntry.getKey());
      }
      newEntry.setKey(key);
      newEntry.setValue(value);
      addNewEntry(newEntry);
      data.put(key, newEntry);
```
</div>
</details>
</div>
<br>


</div>